packages/core/src/index.ts

```diff
@@ -41,7 +41,7 @@ export {
 } from './actors/index.ts';
 
 export { stateIn, not, and, or } from './guards.ts';
-
+export { simulate } from './simulate.ts';
```

Comment:
I'm not super sure about the current shape of this API. I also don't have any concrete feedback about what could be improved here though. I like how we use it internally right now (for the most part), since we can hide some tedious stuff in this lil helper. I'm not sold on exposing it publicly as a stable API though, it feels somewhat 0.1/beta-ish. It would be cool if it could do **more** but at the moment I don't know what that more is. Take it with a grain of salt, it's mostly a feedback based on a vague gut feeling and not on concrete technical aspects. 

It also feels that really this API is mostly useful to machines. Yes, I know that one could maybe implement other logics that would switch their internal implementation to some kind of a "pure variant" when a flag/option/whatever would be passed to that actor logic but that doesn't really sound appealing to me at all. If one wants to simulate a different type of an actor logic then swapping its reference or something to a completely different pure/mocked variant sounds pretty reasonable to me. 

We have a luxury of machines being pure-ish by default and it's super tempting to just reuse the whole thing publicly for both pure and impure use cases but I feel like we are hitting some conceptual limitations with that approach because it's hard to satisfy **both** worlds at the same time. It almost feels to me that `createTestMachine`, with its own types, simplified API, slightly limited but yet operating on the same core concepts... is just a much cleaner solution. 

That doesn't exactly preclude existence of/need for a `simulate`/`simulateMachine` API. Perhaps the use cases of `@xstate/test` and simulation are slightly different? They certainly feel like the same thing but for instance `simulate` should be able to do something with invoked children (even if that thing is just to "record" that they have been created) whereas in `@xstate/test` it doesn't make a lot of sense to have any invokes at all (current `createTestMachine` actually forbids `invoke`).

Reply:
The problem is that if we don't export this, importing them in other package tests gets a lot trickier.

I think we should export this while in beta but not document it. I actually want to get rid of `createTestMachine` because `@xstate/graph` and `@xstate/test` should operate with any behavior, not just a special kind of machine.

Otherwise, what's the best way to move forward with this PR?

Reply:
Ping @Andarist